File: src/main/scala/com/tort/mudai/mapper/Zone.java
package com.tort.mudai.mapper;

public class Zone {
    private String _name;

    public Zone(){

    }

    public Zone(String zoneName){
        _name = zoneName;
    }

    public String getName() {
        return _name;
    }

    public void setName(final String name) {
        _name = name;
    }
}

File: src/main/scala/com/tort/mudai/mapper/Zone.scala
package com.tort.mudai.mapper

import scalaz.{@@, Tag}
import com.tort.mudai.mapper.Zone.ZoneName

class Zone(val id: String, val name: String @@ ZoneName)

object Zone {
  trait ZoneName

  def name(n: String) = Tag[String, ZoneName](n)
}
